src/bin/toolkit.js

```diff
@@ -8,506 +8,309 @@ const fs = require('fs')
-      })
+    return forcedInstallCommand(settings)
   }
-
-  function decodeFilters (mir) {
-    return mir.map((filter) => {
-      if (filter.args.length > 0) {
-        const decodedArgs = cbor.decode(Buffer.from(filter.args))
-        return {...filter, args: decodedArgs}
-      } else {
-        return filter
+}
+
+async function forcedInstallCommand (settings) {
+  return downloadToolkit(
+    settings.paths.toolkitDownloadName,
+    settings.paths.toolkitFileName,
+    settings.paths.toolkitBinPath,
+    settings.system.platform,
+    settings.system.arch
+  )
+    .catch((err) => {
+      console.error(`Error updating witnet_toolkit binary:`, err)
+    })
+}
+
+
+function tasksFromMatchingFiles (args, matcher) {
+  return fs.readdirSync(args[2])
+    .filter((filename) => filename.match(matcher))
+    .map((filename) => [args[0], args[1], path.join(args[2], filename)])
+}
+
+function tasksFromArgs (args) {
+  // Ensure that no task contains arguments starting with `0x`
+  return args.map(arg => arg.replace(/^0x/gm, ''))
+}
+
+async function versionCommand (settings) {
+  return fallbackBinaryCommand(settings, ['--version'])
+}
+
+async function fallbackBinaryCommand (settings, args) {
+  const toolkitOutput = await toolkitRun(settings, args.slice(1))
+    .catch((err) => {
+      let errorMessage = err.message.split('\n').slice(1).join('\n').trim()
+      const errorRegex = /.*^error: (?<message>.*)$.*/gm
+      const matched = errorRegex.exec(err.message)
+      if (matched) {
+        errorMessage = matched.groups.message
       }
+      console.error(errorMessage || err)
     })
+  if (toolkitOutput) console.info(toolkitOutput)
+}
+
+
+/// PROCESS SETTINGS ===============================================================================================
+
+let force;
+let forceIndex = args.indexOf('--force');
+if (forceIndex >= 2) {
+  // If the `--force` flag is found, process it, but remove it from args so it doesn't get passed down to the binary
+  force = args[forceIndex]
+  args.splice(forceIndex, 1)
+}
+
+let help = false
+if (args.includes('--help')) {
+  help = true
+  args.splice(args.indexOf('--help'), 1)
+}
+
+let showVersion = false
+if (args.includes('--version')) {
+  showVersion = true
+  args.splice(args.indexOf('--version'), 1)
+}
+
+let update = false
+if (args.includes('--update')) {
+  update = true
+  args.splice(args.indexOf('--update'), 1)
+}
+
+const settings = {
+  paths: {
+    toolkitBinPath,
+    toolkitDirPath,
+    toolkitDownloadName,
+    toolkitFileName,
+  },
+  checks: {
+    toolkitIsDownloaded,
+  },
+  system: {
+    platform,
+    arch,
+  },
+  force, help, showVersion, update,
+}
+
+
+/// MAIN LOGIC ======================================================================================================
+
+const mainRouter = {
+  '--': fallbackBinaryCommand,
+  'update':  forcedInstallCommand,
+  'install': installCommand,
+  'version': versionCommand,
+}
+
+async function main () {
```

Comment:
Now that we are building a new package, I'd avoid handling the CLI with a custom solution to not reinvent the wheel and use the benefits of a battle-tested tool. A well known library in javascript for this is [commander](https://github.com/tj/commander.js#commanderjs)

Reply:
perhaps for v2.2?